### ISSUE TITLE
Make dynamic require imports return `unkown` 

### ISSUE THREAD

**Issue by rupert648** [total-typescript/ts-reset#55]
Currently, `require` faces a similar problem to `JSON.parse`, where it returns `any`. Like `JSON.parse`, I think this would be another great place to reset the return type to be unkown.

I.e.
```js
var require: NodeRequire (id: string) => any
```
becomes
```js
var require: NodeRequire (id: string) => unknown
```

**Comment by m10rten** [comment]
To be fair: `require` has intellisense on exported instances if I'm not mistaken.

So what exactly is the usecase of making it return an `unknown` type?

**Comment by rupert648** [comment]
> To be fair: `require` has intellisense on exported instances if I'm not mistaken.
> 
> So what exactly is the usecase of making it return an `unknown` type?
> 
> To eleborate:
> 
> When importing like so:
> 
> ```js
> const hi = require("pkg");
> ```
> 
> The variable `hi` will contain all exported stuff and will show it in intellisense, where this may seem as any because it says so, it does not harm the typesafety in my opinion.

Yeah this is true when doing static imports, but if you're using require dynamically, for example
```ts
const importFile = (path: string): any => {
  return require(path);
};
```
The default behaviour of this require is to return `any`.

You're right about the intellisense stuff though so I'll investigate to see if theres a way to test this behaviour is maintained in [the PR](https://github.com/total-typescript/ts-reset/pull/59)

**Comment by m10rten** [comment]
> Yeah this is true when doing static imports, but if you're using require dynamically, for example
> 
> ```ts
> const importFile = (path: string): any => {
>   return require(path);
> };
> ```
> 
> The default behaviour of this require is to return `any`.
> 
> You're right about the intellisense stuff though so I'll investigate to see if theres a way to test this behaviour is maintained in [the PR](https://github.com/total-typescript/ts-reset/pull/59)

I've tested it in JS and no luck, does not change when you return unkown, the wrapper function will also not have any idea what is in there since you put a `path` as import path.

**Comment by rupert648** [comment]
Yep thats exactly what I mean. We can change the Require's default return type to be `unknown` when its given a dynamic string such as in the above example (see [here](https://github.com/total-typescript/ts-reset/pull/59/files#diff-bf4c9f1dff8f7fa682bad50bae190084177f6a055039693e69bf6d33d86d8cfd)), then this function we can change to return `unknown` which in my opinion makes more sense.

When doing top level require, JS's intellisense will still kick in. So this change would only affect the very limited use case of using `require` when not knowing the path being passed in until runtime

**Comment by joshuajaco** [comment]
I think top-level `require` calls should be using the [`import = require()` syntax](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require) anyway, so I think this change should be fine.
